packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1941,6 +1965,20 @@ export class KeyringController extends BaseController<
     return this.#getKeyringById(keyringId);
   }
 
+  /**
+   * Get the metadata for the specified keyring.
+   *
+   * @param keyring - The keyring instance to get the metadata for.
+   * @returns The keyring metadata.
+   */
+  #getKeyringMetadata(keyring: unknown): KeyringMetadata {
```

Comment:
I'm keeping this method as internal because consumers are not supposed to carry keyring instances around